aten/src/ATen/native/README.md

```diff
@@ -286,6 +286,32 @@ In addition to the backends above, we also support the keywords:
     registered to `Math` key should be plain mathematical composition of other
     `at::` functions and support training and inference for any backend.
 
+Note the kernels registered to `DefaultBackend` and `Math` might look identical,
+they only differ in whether it's also used to compute backward as well.
+
+For example, suppose `my_op` can be decomposed in the following way:
+```
+at::Tensor my_op(const Tensor& self, const Tensor& other) {
+  return self + 2 * other;
+}
+```
+
+If we already know inference kernels and derivative formulas for operators `+` and `*` in our system,
+you can just register `my_op` to `Math` and both inference & autograd will just work.
+Although it seems we only write down the inference formula here, PyTorch autograd system would correctly
+set up the backward for `my_op` using the chain formula and derivatives of `+` & `-` operators.
+In other words `d_out/d_self = 1; d_out/d_other = 2` can be derived automatically from
+`my_op` inference kernel. Of course if we don't have derivative formula defined for any of `+` or `-`,
+backward of `my_op` can no longer be derived automatically.
+
+Whether to use `Math` or `DefaultBackend` for your kernel can be decided by the following steps:
+1. Always start with a `Math` kernel that's composable of other existing operators.
```

Comment:
s/composable/composed/

Reply:
Also, shouldn't we say "if you can" here? I.e. are we considering the case where they can do a backend-agnostic kernel but they're calling things that don't support autograd? 

If so, then we should so add some "if you don't want to or can't" type language to the next items as well.